index.html

```diff
@@ -564,8 +564,9 @@ <h4>
             NFC Forum <dfn>local type</dfn> that are defined by the NFC Forum or
             by an application, within the scope of the local context of another
             record or application and always start with lowercase letter or a
-            number. See <a>Smart poster</a> for an example on how local types are
-            used.
+            number. Those are usually short to save on potentially long domain-name
```

Comment:
> Those are usually short to save on potentially long domain-name containing record types.

Local types are usually short strings that are unique in the local context of the containing record. They are used when it's not worth defining full external types (that might contain long domain names) and when there is no need to define the meaning of the types outside of the local context of the containing record.

Reply:
Local types are string defined type names that are unique only within the local context of the containing record. As these are contained within another record, there is no need for adding a developer owned domain as namespace as required by external types, which takes up additional storage, as their meaning within the record should be known if the containing record type is known.

Reply:
What do you think of the new patch?

Reply:
Looks good. Next, perhaps we also need to add precision to algorithms wrt the external and local types parsing/validation. We should require FQDN in external types, then we'd need a pointer to an external algorithm on parsing/validating FQDN (I know there are a number of reg-exps around). Also, perhaps we need to factor out an algo for validating local types (check for local context and case).